keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+devices and frequent status updates. In large-scale clusters where many devices
+are allocated, this impact must be considered.
+
+## Design Details
+
+### API
+
+The `ResourceClaimStatus` struct in `pkg/apis/resource/types.go` will be
+extended to include the slice of `Devices`.
+
+`ResourceClaim` validation of the status in
+`pkg/apis/resource/validation/validation.go` will be covered to allow a device
+to be reported only once in the slice, a device is being identified  by
+`<driver name>/<pool name>/<device name>`.
+
+### Write Permission
+
+To prevent unauthorized or accidental modifications by entities that do not 
+have access to a particular resource, a `ValidatingAdmissionPolicy` will be 
+created to validate the entities attempting to update the devices in the 
+`ResourceClaim.Status`.
+
+The `ValidatingAdmissionPolicy` will restrict `ResourceClaim.Status.Devices` 
+to be set only during updates, as the object will have first to be created and 
+allocated, then configured inside the pods. It will also restrict the 
+`ResourceClaim.Status.Devices` to be set only for when the `ResourceClaim` is 
+allocated to a node. Additionally, the allocated node where the `ResourceClaim` 
+is assigned will be used to check if the user/entity updating the 
+`ResourceClaim.Status.Devices` is running on the same node.
+
+The allocated node for the `ResourceClaim` must be unique in the form of 
+`nodeSelector.nodeSelectorTerms[0].matchFields[0].values[0] = <NodeName>` and
+`nodeSelector.nodeSelectorTerms[0].matchFields[0].key = metadata.name` and
+`nodeSelector.nodeSelectorTerms[0].matchFields[0].operator = In`. Without this
+unique format, the update of the devices in the status will be rejected.
+
+Here is a `ResourceClaim` allocated on a node. This would only work for now if 
+exactly one node is set:
+```yaml
+apiVersion: resource.k8s.io/v1alpha3
+kind: ResourceClaim
+metadata:
+  ...
+spec:
+  ...
+status:
+  allocation:
+    ...
+    nodeSelector:
+      nodeSelectorTerms:
+      - matchFields:
+        - key: metadata.name
+          operator: In
+          values:
+          - my-node
+  ...
+```
+
+Here is an example of how the `ValidatingAdmissionPolicy` could look like:
+```yaml
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "resourceclaim-device-status-update"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups: ["resource.k8s.io"]
+      apiVersions: ["*"]
+      operations: ["UPDATE"]
+      resources: ["resourceclaims"]
+  matchConditions:
+    - name: 'device-status-update'
+      expression: >- # Validation only for objects with their .status.devices updated.
+        object.status.devices != oldObject.status.devices
+  validations:
+  - expression: >- # User node must be the same node as the one where the ResourceClaim is allocated.
+      variables.userNodeName != variables.objectNodeName
```

Comment:
Aren't VAP's asserting the affirmative?  I.e. this should be `==`  ?  If not, I am confused about my own VAPs.

Reply:
I will re-try this VAP with my PoC,`validations` should work fine, but I remember having issues with the `matchConditions`